faunadb-common/src/main/java/com/faunadb/common/Connection.java

```diff
@@ -490,4 +520,36 @@ private void logFailure(HttpRequest request, Throwable ex) {
   private static String generateAuthHeader(String authToken) {
     return "Bearer " + authToken;
   }
+
+  private static String getRuntimeEnv() {
+    if (System.getenv("NETLIFY_IMAGES_CDN_DOMAIN") != null) {
+      return "Netlify";
+    }
+
+    if (System.getenv("VERCEL") != null) {
+      return "Vercel";
+    }
+
+    if (System.getenv("PATH") != null && System.getenv("PATH").contains(".heroku")) {
+      return "Heroku";
+    }
+
+    if (System.getenv("AWS_LAMBDA_FUNCTION_VERSION") != null) {
+      return "AWS Lambda";
+    }
+
+    if (System.getenv("_") != null && System.getenv("_").contains("google")) {
+      return "GCP Cloud Functions";
+    }
+
+    if (System.getenv("GOOGLE_CLOUD_PROJECT") != null) {
+      return "GCP Compute Instances";
+    }
+
+    if (System.getenv("ORYX_ENV_TYPE") != null && System.getenv("WEBSITE_INSTANCE_ID") != null && System.getenv("ORYX_ENV_TYPE") == "AppService") {
+      return "Azure Compute";
+    }
+
+    return "Unknown";
```

Comment:
Maybe this could be an `Option` instead.

Reply:
return `unknown` for non-recognized env is requirement